lib/Sema/CSApply.cpp

```diff
@@ -2487,7 +2487,14 @@ namespace {
             if (member->isInstanceMember()) {
               return;
             }
-            
+
+            // If we have something like 'static let none = 1', then ignore.
+            if (auto VD = dyn_cast<VarDecl>(member)) {
+              if (!VD->getInterfaceType()->isEqual(
+                      baseTyNominalDecl->getDeclaredInterfaceType()))
+                return;
+            }
+
```

Comment:
While you're here, what happens if there's a static *method* named `none`? Seems like that's similar to the enum-with-associated-values case.

Reply:
And what if there's both an instance member and a static member named `none`?

Reply:
We don't include instance members in the check, only static members. I suppose for the method case, we can check the result type of the method:

```swift
enum Foo {
  case bar
  static func none() -> Foo { return Foo.bar }
}

let _: Foo? = .none // warning
```

Reply:
Obviously, the warning has to be tweaked, so it mentions calling `Foo.none()`

Reply:
> We don't include instance members in the check

The current implementation early-exits if it sees an instance member, even if there's also a static member. That's why I brought it up.

I'm okay with the method case just being skipped too; the point is that it's *not* ambiguous (though as Brent points out the criterion is "confusing" more than "ambiguous").

Reply:
Oh, we already diagnose (even without this patch):

```
enum Foo {
  case bar
  static let none = Foo.bar
  var none: Foo { return .bar }
}

let optionalFoo: Foo? = .none // warning
```

Reply:
Ah of course, it depends on source order 😅 That should be fixed!

Reply:
Fixed.